### ISSUE TITLE
Try to respect viewhint in Tables.subset more often

### ISSUE THREAD

**Issue by bkamins** [JuliaData/Tables.jl#312]
Currently lines
https://github.com/JuliaData/Tables.jl/blob/2cb13998e856692ed273c931b83477caf8b7b020/src/Tables.jl#L609
and
https://github.com/JuliaData/Tables.jl/blob/2cb13998e856692ed273c931b83477caf8b7b020/src/Tables.jl#L618
ignore `viewhint` keyword argument. This is OK, but maybe it would be possible to change them in a way that they could respect it (at least in some cases).

**Comment by quinnj** [comment]
I'm not sure I follow. In the `inds isa Integer` case, I view this as just a normal `x[i]` indexing operation, which just returns the object at the specified index. Are you proposing we make a copy of the row if `viewhint=false`?

**Comment by bkamins** [comment]
This is a tricky thing to discuss. I can explain you what @ablaom found non-intuitive on DataFrames.jl example (which works correctly):

```
julia> using DataFrames

julia> df = DataFrame(a=1)
1×1 DataFrame
 Row │ a
     │ Int64
─────┼───────
   1 │     1

julia> Tables.subset(df, 1, viewhint=true) # returned object is a view into a source table, so modifying table changes the values in this object
DataFrameRow
 Row │ a
     │ Int64
─────┼───────
   1 │     1

julia> Tables.subset(df, 1, viewhint=false) # returned object is not a view into a source table, so modifying table does not change the values in this object
(a = 1,)
```

In essence - it is not an issue if it is a copy or not (we do not care). What @ablaom cares about is that when you get a row with `viewhint=true` then modifying the source table should update the row, while if `viewhint=false` is passed then the object is detached.

The issue is that for some tables it is not even possible to implement these two approaches (and that is why it is `viewhint` not just `view` - as we discussed previously). For other tables- you will not know if `x[i]` returns you a detached object or not. So the issue is hard here.

The question is - if we can in more cases follow the `viewhint` kwarg value. Of course - the ideal situation is that if some type declares it is a table it should implement its own `Tables.subset` method (like I did in DataFrames.jl) to make sure a proper handling is done.
